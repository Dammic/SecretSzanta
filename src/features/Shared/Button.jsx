import React from 'react'
import PropTypes from 'prop-types'

export const Button = ({ label, onClick }) => {
    return (
        <div
            className="fabulous-button lined thick"
            onClick={onClick}
            role="button"
            tabIndex="0"
        >
            {label}
        </div>
    )
}

Button.propTypes = {
    label: PropTypes.string,
    onClick: PropTypes.func,
}
