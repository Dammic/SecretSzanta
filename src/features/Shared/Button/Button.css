@import 'variables/variables.css';

.fabulousButton {
    background: transparent;
    padding: 1rem 1rem;
    margin: 0 1rem;
    transition: all .5s ease;
    color: #41403E;
    font-size: 2rem;
    letter-spacing: 1px;
    outline: none;
    box-shadow: 20px 38px 34px -26px hsla(0,0%,0%,.2);
    border-radius: 255px 15px 225px 15px/15px 225px 15px 255px;
    &:hover {
        box-shadow:2px 8px 4px -6px hsla(0,0%,0%,.3);
        background-color: var(--fabulous-button-hover-bg);
        cursor: pointer;
    }
}
.linedThick {
    border: solid 7px var(--fabulous-button-border);
}
.dottedThick {
    border: dotted 5px var(--fabulous-button-border);
}
.dashedThick {
    border: dashed 5px var(--fabulous-button-border);
}
.linedThin {
    border: solid 2px var(--fabulous-button-border);
}
.dottedThin {
    border: dotted 2px var(--fabulous-button-border);
}
.dashedThin {
    border: dashed 2px var(--fabulous-button-border);
}
