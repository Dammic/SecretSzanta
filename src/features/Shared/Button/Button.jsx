import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './Button.css'

export const Button = ({ label, onClick, className }) => {
    return (
        <div
            className={classNames(styles.fabulousButton, styles.linedThick, className)}
            onClick={onClick}
            role="button"
            tabIndex="0"
        >
            {label}
        </div>
    )
}

Button.propTypes = {
    label: PropTypes.string,
    onClick: PropTypes.func,
    className: PropTypes.string,
}
