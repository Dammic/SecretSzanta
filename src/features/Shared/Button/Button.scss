.fabulous-button {
    background: transparent;
    padding: 1rem 1rem;
    margin: 0 1rem;
    transition: all .5s ease;
    color: #41403E;
    font-size: 2rem;
    letter-spacing: 1px;
    outline: none;
    box-shadow: 20px 38px 34px -26px hsla(0,0%,0%,.2);
    border-radius: 255px 15px 225px 15px/15px 225px 15px 255px;
    &:hover{
        box-shadow:2px 8px 4px -6px hsla(0,0%,0%,.3);
        background-color: $fabulous-button-hover-bg;
        cursor: pointer;
    } 
    &.lined.thick{
        border: solid 7px $fabulous-button-border;
    }
    &.dotted.thick{
        border: dotted 5px $fabulous-button-border;
    }
    &.dashed.thick{
        border: dashed 5px $fabulous-button-border;
    }
    &.lined.thin{
        border: solid 2px $fabulous-button-border;
    }
    &.dotted.thin{
        border: dotted 2px $fabulous-button-border;
    }
    &.dashed.thin{
        border: dashed 2px $fabulous-button-border;
    }
}
