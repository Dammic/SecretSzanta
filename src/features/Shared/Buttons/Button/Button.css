@import 'variables/variables.css';

.btn {
    background-color: var(--button-default-bg);
    display: inline-flex;
    align-items: center;
    justify-content: center;
    color: var(--button-default-font-color);
    transition: .05s ease-in;
    font-size: 14px;
    padding: 2px 5px;
    border-radius: 0;
    border: 1px solid var(--button-default-border);
    text-transform: capitalize;
    transition: 0.1s linear;

    &:hover {
        background-color: var(--button-default-bg-hover);
	border-color: var(--button-default-border-hover);
        cursor: pointer;
    }

    &:active {
        background-color: var(--button-default-bg-active);
        border-color: var(--button-default-border-active);
    }
}
