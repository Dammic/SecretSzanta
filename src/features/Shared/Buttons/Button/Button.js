import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './Button.css'

const Button = (props) => {
    const { className, children } = props
    return (
        <div
            role="button"
            tabIndex="0"
            {...props}
            className={classNames(styles.btn, className)}
        >
            {children}
        </div>
    )
}

Button.displayName = 'Button'
Button.propTypes = {
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    className: PropTypes.string,
}

export default Button
