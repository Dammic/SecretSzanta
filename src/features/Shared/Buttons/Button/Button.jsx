import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './Button.css'

class Button extends PureComponent {
    render() {
        const { onClick, className, children } = this.props
        return (
            <div
                className={classNames(styles.btn, className)}
                onClick={onClick}
                role="button"
                tabIndex="0"
            >
                {children}
            </div>
        )
    }
}

Button.propTypes = {
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    onClick: PropTypes.func,
    className: PropTypes.string,
}

export default Button
