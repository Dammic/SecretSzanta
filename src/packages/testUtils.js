import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
