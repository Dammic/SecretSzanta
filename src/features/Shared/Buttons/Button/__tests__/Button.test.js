import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from 'packages/testUtils'
import Button from '../Button'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        children: <span>mockLabel</span>,
        className: 'mockClassName',
        ...propsOverrides,
    }
    const component = renderMethod(<Button {...props} />)
    return { props, component }
}

describe('<Button />', () => {
    it('matches snapshot', () => {
        const { props } = setupProps()
        expectMatchingSnapshot(<Button {...props} />)
    })
})
