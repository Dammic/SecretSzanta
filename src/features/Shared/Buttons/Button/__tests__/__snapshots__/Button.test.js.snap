// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<div
  className="btn mockClassName"
  role="button"
  tabIndex="0"
>
  <span>
    mockLabel
  </span>
</div>
`;
